winpr/libwinpr/utils/ini.c

```diff
@@ -304,35 +322,45 @@ static wIniFileSection* IniFile_GetSection(wIniFile* ini, const char* name)
 	return section;
 }
 
+static BOOL IniFile_SectionResize(wIniFile* ini, size_t count)
+{
+	WINPR_ASSERT(ini);
+
+	if (ini->nSections + count >= ini->cSections)
+	{
+		const size_t new_size = ini->cSections + count + 1024;
```

Comment:
Why is the `+ 1024` (same in `IniFile_SectionKeysResize`)?
It seems we are always carrying 1024 additional empty slots around with us

Reply:
block allocation, always extend by `1024` if we reach the current limit